python/graphstorm/model/hgt_encoder.py

```diff
@@ -0,0 +1,288 @@
+                                                                            self.num_heads,
+                                                                            self.d_k)
+                else:
+                    q_val = q_linear(h[dsttype]).view(-1, self.num_heads, self.d_k)
+
+                relation_att = self.relation_att[c_etype_str]
+                relation_pri = self.relation_pri[c_etype_str]
+                relation_msg = self.relation_msg[c_etype_str]
+
+                k_val = torch.einsum("bij,ijk->bik", k_val, relation_att)
+                v_val = torch.einsum("bij,ijk->bik", v_val, relation_msg)
+
+                sub_graph.srcdata['k'] = k_val
+                sub_graph.dstdata['q'] = q_val
+                sub_graph.srcdata[f'v_{c_etype_str}'] = v_val
+
+                sub_graph.apply_edges(fn.v_dot_u('q', 'k', 't'))
+                attn_score = sub_graph.edata.pop('t').sum(-1) * relation_pri / self.sqrt_dk
+                attn_score = edge_softmax(sub_graph, attn_score, norm_by='dst')
+                sub_graph.edata[f't_{c_etype_str}'] = attn_score.unsqueeze(-1)
+
+            edge_fn = {}
+            for srctype, etype, dsttype in g.canonical_etypes:
+                c_etype_str = '_'.join((srctype, etype, dsttype))
+                edge_fn[srctype, etype, dsttype] = (fn.u_mul_e(f'v_{c_etype_str}',
+                                                               f't_{c_etype_str}', 'm'),
+                                                    fn.sum('m', 't'))
+
+            g.multi_update_all(edge_fn, cross_reducer="mean")
+
+            new_h = {}
+            for k, _ in h.items():
+                if g.num_dst_nodes(k) > 0:
+                    alpha = torch.sigmoid(self.skip[k])
+                    if g.dstnodes[k].data.get('t') is not None:
+                        t = g.dstnodes[k].data['t'].view(-1, self.out_dim)
+                        trans_out = self.drop(self.a_linears[k](t))
+                        if g.is_block:
+                            trans_out = trans_out * alpha + h[k][:g.num_dst_nodes(k)] * (1-alpha)
+                        else:
+                            trans_out = trans_out * alpha + h[k] * (1-alpha)
+                    else:                       # Nodes not really in destination side.
+                        warnings.warn("Warning. Graph convolution returned empty "
+                          f"dictionary, for node with type: {str(k)}")
+                        # So add psudo self-loop with feature copy.
+                        trans_out = self.drop(self.a_linears[k](h[k]))
+                        trans_out = trans_out * alpha + h[k] * (1-alpha)
+                else:
+                    continue
+
+                if self.use_norm:
+                    new_h[k] = self.norms[k](trans_out)
+                else:
+                    new_h[k] = trans_out
+
+                if self.num_ffn_layers_in_gnn > 0:
+                    new_h[k] = self.ngnn_mlp(new_h[k])
```

Comment:
Follow the rgcn_encoder, rgat_encoder, and sage_encoder, added another activation function before do ngnn_mlp so that we have at least an activation function.